### PR TITLE
feat:add a new filed EnableRating in repository

### PR DESCRIPTION
**Pull request by zqq454224016**
#### Which issue(s) this PR fixes

Fixes #345

### CODE REVIEW

File: config/samples/rating_test.sh [hunk 1]
```diff
@@ -148,6 +148,12 @@ function waitRatingDone() {
 	ratingName=$2
 	START_TIME=$(date +%s)
 	sleep 2 # wait for operator patch status. avoid 0=0 situations
+	result=$(kubectl -n${namespace} get rating ${ratingName} -ojson --ignore-not-found=true | jq -r '.status.conditions[0].status')
```
Comment (bjwswang) [inline config/samples/rating_test.sh:151]: You should check
`type`  ==   `Ready`
`Status` ==  `False`
`Message` == `Rating disabled in component's repository` 
in another function `waitRatingFail` with same `while true` loop chekcing.

**Review by bjwswang — CHANGES_REQUESTED** [review]
@zqq454224016  Good job! Some tiny changes still required when doing the example-test.

**Review by bjwswang — DISMISSED** [review]

File: controllers/rating_controller.go [hunk 2]
```diff
@@ -110,6 +111,27 @@ func (r RatingReconciler) ratingChecker(ctx context.Context, instance *corev1alp
 	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.ComponentName}, &component); err != nil {
 		return false, err
 	}
+
+	repository := corev1alpha1.Repository{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: component.Status.RepositoryRef.Name}, &repository); err != nil || !repository.Spec.EnableRating {
+		instanceDeepCopy := instance.DeepCopy()
+		cond := corev1alpha1.Condition{
+			Status:             v1.ConditionFalse,
```
Comment (Abirdcfly) [inline controllers/rating_controller.go:119]: So rating's condition is only show when there are some error or rating is not enable?

Comment (bjwswang) [reply controllers/rating_controller.go:119]: We should update condition when
1. error is not nil and  not `NotFound`
2. error is nil, but rating is disabled. 

@zqq454224016

Comment (zqq454224016) [reply controllers/rating_controller.go:122]: When the rating is disable, what error is returned?

Comment (bjwswang) [reply controllers/rating_controller.go:122]: A simple error message is enough just like the condition message you used here.

Comment (bjwswang) [reply controllers/rating_controller.go:122]: > So rating's condition is only show when there are some error or rating is not enable?

Actually we should track the whole running status in Rating.Let's do this in another pr

File: config/samples/rating_test.sh [hunk 3]
```diff
@@ -147,7 +147,7 @@ function waitRatingDone() {
 	namespace=$1
 	ratingName=$2
 	START_TIME=$(date +%s)
-	sleep 2 # wait for operator patch status. avoid 0=0 situations
+	sleep 2 # wait for operator patch status. avoid 0=0 situationss
```
Comment (Abirdcfly) [inline config/samples/rating_test.sh:150]: 😳？

**Review by Abirdcfly — APPROVED** [review]

**Review by bjwswang — APPROVED** [review]

**Pull request merged**